tests/Content/PlainTextStorageTest.php

```diff
@@ -378,19 +377,6 @@ public function testReadLatestSingleLang()
 		$this->assertSame($fields, $this->storage->read(VersionId::latest(), Language::single()));
 	}
 
-	/**
-	 * @covers ::read
-	 */
-	public function testReadWhenMissing()
-	{
-		$this->setUpSingleLanguage();
-
-		$this->expectException(NotFoundException::class);
-		$this->expectExceptionMessage('Version "changes" does not already exist');
-
-		$this->storage->read(VersionId::changes(), Language::single());
-	}
```

Comment:
`Question:` What's now returned when calling `$storage->read()` when non existent?

Reply:
The storage can throw its own exception in this case. But since we wrap the storage methods anyway and you never interact with them directly, the same exception is simply thrown one level up before ->storage()->read() is even called.

Reply:
I understand how it is working in our use case with version. But I think we need to define what's the expected case for the storage handler method here (return null, exception, ?) and then have a test for it. Cause with removing the test completely, we leave it unsure what's the behavior of a storage handler for this case.

Reply:
I think that the type hinting in the read method is also indicating that the storage handler should always return an array. If it can't provide that array, it can throw an exception to indicate that something went wrong.

Reply:
> I think that the type hinting in the read method is also indicating that the storage handler should always return an array. 

Exactly, but ours currently don't, e.g. `MemoryStorage::read()` runs into an error as it returns `null` but type hints for an array, while `PlainTextStorage::read()` returns an empty array for the same situation. I don't think we can leave it like this but need to ensure a consistent behavior and also have tests for that.

Reply:
Ok, I went over the code again and removed the Exception annotations and decided to keep MemoryStorage and PlainTextStorage consistent by always returning an array. 

We have `exists` here to run additional checks if necessary and otherwise it's probably the easiest and cleanest option to just always expect an array